lib/livebook/app.ex

```diff
@@ -0,0 +1,119 @@
+if Mix.target() == :app do
+  defmodule Livebook.App do
+    @moduledoc false
+
+    @behaviour :wx_object
+
+    # https://github.com/erlang/otp/blob/OTP-24.1.2/lib/wx/include/wx.hrl#L1314
+    @wx_id_exit 5006
+
+    def start_link(_) do
+      {:wx_ref, _, _, pid} = :wx_object.start_link(__MODULE__, [], [])
+      {:ok, pid}
+    end
+
+    def child_spec(init_arg) do
+      %{
+        id: __MODULE__,
+        start: {__MODULE__, :start_link, [init_arg]},
+        restart: :transient
+      }
+    end
+
+    @impl true
+    def init(_) do
+      title = "Livebook"
+
+      wx = :wx.new()
+      frame = :wxFrame.new(wx, -1, title, size: {0, 0})
+
+      if macos?() do
+        fixup_macos_menubar(frame, title)
+      end
+
+      :wxFrame.show(frame)
+      :wxFrame.connect(frame, :command_menu_selected)
+      :wxFrame.connect(frame, :close_window, skip: true)
+      :wx.subscribe_events()
+      state = %{frame: frame}
+
+      Livebook.Utils.browser_open(LivebookWeb.Endpoint.access_url())
```

Comment:
Did you find this has any advantage over `:wx_misc.launchDefaultBrowser/1` or do you just want to avoid additional `:wx` dependencies?

Reply:
Oh, didn't know about that one! Yeah we'd rather avoid depending on wx because we use it  in non-desktop code: `$ livebook server --open`